### PR TITLE
Handle SizeLimit error

### PR DESCRIPTION
**Pull request by shusann01116**
This solves #2481. Handle if `AsyncStorage::fetch_source_file` returnes `SizeLimitReached` error.

The message shown in the page will be like the screen shot.

image

### CODE REVIEW

**Review by syphar — CHANGES_REQUESTED** [review]
Awesome, thank you! 

Two changes would be awesome: 
- cold you move the warning to the right of the file list? 
- also do you mind adding a small test? something like `fetch_source_file_utf8_path` is probably enough, checking for the existance of the warning. 

I'm not 100% certain if we should change the error returned by the `SizedBuffer` to make the matching easier, but it's ok for now.

**Comment by shusann01116** [comment]
Thank you for the review!
I've made changes as you mentioned.

## Moved the warning to the right of the file list

image

## Added a tiny test to ensure the message exists

I'm not sure, but I mocked a file has 50MB + 1Byte array to exceeds the default limit.

File: src/web/source.rs [hunk 1]
```diff
@@ -729,4 +737,23 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    fn large_file_test() {
+        wrapper(|env| {
```
Comment (syphar) [inline src/web/source.rs:743]: you can change the config for the test, so you don't have to create a 50 MB file in memory
```suggestion
        wrapper(|env| {
            env.override_config(|config| {
                config.max_file_size = 1;
                config.max_file_size_html = 1;
            });
            // ... 

```

**Review by syphar — CHANGES_REQUESTED** [review]
last remaining bit, then we can merge and deploy! 

I'm already looking forward to deploying the change and reducing error logs :)

File: src/web/source.rs (hunk 1 above)
Comment (shusann01116) [reply src/web/source.rs:743]: Thank you!
Revised the code.

**Review by syphar — APPROVED** [review]

**Pull request merged**